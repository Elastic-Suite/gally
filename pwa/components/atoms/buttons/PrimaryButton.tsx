import Button from '@mui/material/Button'
import CommonButtonStyle from "~/components/atoms/buttons/CommonButtonStyle";

const PrimaryButton = (props) => {
  return (
    <Button {...props}
      sx={CommonButtonStyle}
      variant="contained"
    >
      {props.children}
    </Button>
  )
}

export default PrimaryButton
