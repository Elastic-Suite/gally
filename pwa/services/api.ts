import { apiUrl } from '~/constants'

export function fetchApi<T>(
  language: string,
  url: RequestInfo | URL,
  options: RequestInit = {}
): Promise<T> {
  if (process.env.NEXT_PUBLIC_LOCAL) {
    if (!String(url).startsWith('/')) {
      url = `/${url}`
    }
    if (!String(url).endsWith('.json') && !String(url).endsWith('.jsonld')) {
      url = `${url}.json`
    }
    url = `/mocks${url}`
  }
  url = `${apiUrl}${url}`
  return fetch(url, {
    ...options,
    headers: {
      ...options.headers,
      'Accept-Language': language,
    },
  }).then((response) => response.json())
}
