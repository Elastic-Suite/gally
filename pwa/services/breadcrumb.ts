import { IMenuChild } from '~/store'

export function getSlugArray(data: string[] | string): string[] {
  const newBreadCrumbData = [data[0]]
  for (let index = 0; index < data.length - 1; index++) {
    newBreadCrumbData.push(`${newBreadCrumbData[index]}_${data[index + 1]}`)
  }
  return newBreadCrumbData
}

export function findBreadcrumbLabel(
  findIndex: number,
  slug: string[],
  menu: IMenuChild[],
  deepIndex = 0
): string {
  if (typeof menu === 'object') {
    for (const menuChild in menu) {
      if (menu[menuChild].code === slug[deepIndex]) {
        if (findIndex === deepIndex) {
          return menu[menuChild].label
        }
        return findBreadcrumbLabel(
          findIndex,
          slug,
          menu[menuChild].children,
          deepIndex + 1
        )
      }
    }
  }
  return null
}
