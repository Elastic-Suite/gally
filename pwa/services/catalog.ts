import { ICatalog } from '~/types'

export function findDefaultCatalog(catalogsData: ICatalog[]): ICatalog | null {
  const defaultCatalog = catalogsData
    ? catalogsData.filter((catalog) =>
        catalog.localizedCatalogs.find((localizedCtl) => localizedCtl.isDefault)
      )[0]
    : null

  if (defaultCatalog) {
    defaultCatalog.localizedCatalogs = defaultCatalog.localizedCatalogs.filter(
      (localizedCtl) => localizedCtl.isDefault
    )
  }
  return defaultCatalog ? defaultCatalog : null
}
