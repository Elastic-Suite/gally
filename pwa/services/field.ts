import { IElasticSuiteProperty, IField } from '~/types'

function updateProperties(
  properties: IElasticSuiteProperty,
  key: string,
  value: number | boolean
): IElasticSuiteProperty {
  if (key === 'visible') {
    return {
      ...properties,
      visible: value as boolean,
    }
  } else if (key === 'editable') {
    return {
      ...properties,
      editable: value as boolean,
    }
  } else if (key === 'position') {
    return {
      ...properties,
      position: value as number,
    }
  }
  return properties
}

export function updatePropertiesAccordingToPath(
  field: IField,
  path: string
): IField {
  if (path.includes('admin/settings')) {
    path = 'settings_attribute'
  } else {
    path = path.replaceAll('/', '_').replace('_admin_', '')
  }
  if (field.elasticsuite?.context) {
    const [_, newPropertiesvalues] = Object.entries(field.elasticsuite?.context)
      .filter(([key, _]) => key === path)
      .flat()
    if (newPropertiesvalues) {
      let newProperties = field.elasticsuite
      // eslint-disable-next-line no-return-assign
      Object.entries(newPropertiesvalues).forEach(
        ([property, propertyValue]) =>
          (newProperties = updateProperties(
            newProperties,
            property,
            propertyValue
          ))
      )
      return {
        ...field,
        elasticsuite: newProperties,
      }
    }
  }
  return field
}
