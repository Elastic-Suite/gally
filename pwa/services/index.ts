export * from './api'
export * from './authProvider'
export * from './format'
export * from './hydra'
export * from './jsonld'
export * from './local'
export * from './table'
export * from './tests'
