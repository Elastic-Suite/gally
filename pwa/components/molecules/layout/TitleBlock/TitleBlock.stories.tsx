import { ComponentMeta, ComponentStory } from '@storybook/react'
import TitleBlock from './TitleBlock'
import PrimaryButton from '../../../atoms/buttons/PrimaryButton'

export default {
  title: 'Atoms/TitleBlock',
  component: TitleBlock,
} as ComponentMeta<typeof TitleBlock>

const Template: ComponentStory<typeof TitleBlock> = (args) => (
  <TitleBlock {...args}>
    <PrimaryButton disabled={false} endIcon="" size="medium" startIcon="">
      Hello world
    </PrimaryButton>
  </TitleBlock>
)

export const Default = Template.bind({})
Default.args = {
  title: 'Hello',
}
