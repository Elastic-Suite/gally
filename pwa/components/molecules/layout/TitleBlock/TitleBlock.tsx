import { styled } from '@mui/system'
import { ReactChild } from 'react'

import { getCustomScrollBarStyles } from '~/services'

const CustomTitle = styled('div')(({ theme }) => ({
  color: theme.palette.colors.neutral[900],
  fontWeight: 600,
  fontSize: '14px',
  lineHeight: '20px',
  fontFamily: 'Inter',
  padding: theme.spacing(2),
  borderBottom: '1px solid',
  borderColor: theme.palette.colors.neutral[300],
}))

const Container = styled('div')(({ theme }) => ({
  borderBottom: '1px solid',
  borderColor: theme.palette.colors.neutral[300],
  '&:last-of-type': {
    borderBottom: 'none',
  },
  overflow: 'auto',
  ...getCustomScrollBarStyles(theme),
}))

const PaddingBox = styled('div')(({ theme }) => ({
  display: 'inline-block',
  padding: theme.spacing(2),
}))

interface IProps {
  title: string
  children: ReactChild
}

function TitleBlock({ title, children }: IProps): JSX.Element {
  return (
    <>
      <CustomTitle>{title}</CustomTitle>
      <Container>
        <PaddingBox>{children}</PaddingBox>
      </Container>
    </>
  )
}

export default TitleBlock
