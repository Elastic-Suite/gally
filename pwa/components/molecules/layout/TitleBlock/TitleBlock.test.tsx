import TitleBlock from './TitleBlock'
import { renderWithProviders } from '~/services'
import PrimaryButton from '../../../atoms/buttons/PrimaryButton'

describe('TitleBlock match snapshot', () => {
  it('PageTitle simple', () => {
    const { container } = renderWithProviders(
      <TitleBlock title="Hello World">Hello</TitleBlock>
    )
    expect(container).toMatchSnapshot()
  })

  it('PageTitle with Composant on Children', () => {
    const { container } = renderWithProviders(
      <TitleBlock title="Hello World">
        <PrimaryButton disabled={false} endIcon="" size="medium" startIcon="">
          Hello world
        </PrimaryButton>
      </TitleBlock>
    )
    expect(container).toMatchSnapshot()
  })
})
