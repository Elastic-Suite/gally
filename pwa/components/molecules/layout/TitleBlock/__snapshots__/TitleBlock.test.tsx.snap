// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBlock match snapshot PageTitle simple 1`] = `
<div>
  <div
    class="css-1sl3yy7"
  >
    Hello World
  </div>
  <div
    class="css-1fhr9tz"
  >
    Hello
  </div>
</div>
`;

exports[`TitleBlock match snapshot PageTitle with Composant on Children 1`] = `
<div>
  <div
    class="css-1sl3yy7"
  >
    Hello World
  </div>
  <div
    class="css-1fhr9tz"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-13u4ebv-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      
      Hello world
      
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
